File: src/main/java/com/example/option/None.java
package com.example.option;

/** Represents the absence of a value. */
public final class None<T> implements Option<T> {
    public None() {
    }

    @Override
    public boolean isSome() {
        return false;
    }

    @Override
    public T get() {
        return null;
    }
}

File: src/main/node/magma/option/None.ts
import Iter from "../list/Iter";
/** Represents the absence of a value. */
export default class None<T> implements Option<T> {
    None(): public {
        // TODO
    }

    @Override
    isSome(): boolean {
        return /* TODO */;
    }

    @Override
    get(): T {
        return /* TODO */;
    }

    @Override
    toIter(): Iter<T> {
        return new OptionIter<T>(this);
    }
}
